projects/drivers/axi_adc_core/axi_adc_core.c

```diff
@@ -88,6 +88,24 @@ int32_t axi_adc_set_pnsel(struct axi_adc *adc,
 	return SUCCESS;
 }
 
+/***************************************************************************//**
+ * @brief axi_adc_get_sampling_freq
+*******************************************************************************/
+int32_t axi_adc_get_sampling_freq(struct axi_adc *adc,
+				  uint32_t chan,
+				  uint64_t *sampling_freq)
+{
+	uint32_t freq;
+	uint32_t ratio;
+
+	axi_adc_read(adc, AXI_ADC_REG_CLK_FREQ, &freq);
```

Comment:
can't we check the return of this functions ?

Reply:
Since no other function is doing it, this can be the subject of a future patch.